compiler/rustc_builtin_macros/src/alloc_error_handler.rs

```diff
@@ -36,7 +36,7 @@ pub fn expand(
         };
 
     // Generate a bunch of new items using the AllocFnFactory
-    let span = ecx.with_def_site_ctxt(item.span);
+    let span = ecx.with_def_site_ctxt(span);
```

Comment:
@oli-obk this is the change that causes the change for alloc_error_handler. I would honestly keep it this way, given how niche it is (it's not gonna be encountered by a newcomer, no matter how intrepid they might be).